torch/_higher_order_ops/associative_scan.py

```diff
@@ -153,18 +220,20 @@ def add(x: torch.Tensor, y: torch.Tensor):
     if reverse:
         leaves = [torch.flip(elem, [dim]) for elem in leaves]
 
-    shape = leaves[0].shape
-    ndim = len(shape)
-    dim = utils.canonicalize_dim(ndim, dim)
+    # Move scan dim to 0 and always perform scan on dim 0
+    orig_scan_dim = dim
+    leaves = [shift_source_dim_to_target_dim(elem, int(dim), 0) for elem in leaves]
```

Comment:
we might replace `shift_source_dim_to_target_dim` with torch.movedim(elem, int(dim), 0).